Assets/MixedRealityToolkit/Utilities/StandardShader/BaseMeshOutline.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// Abstract component to encapsulate common functionality around outline components. 
+    /// </summary>
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/README_MRTKStandardShader.html#mesh-outlines")]
+    public abstract class BaseMeshOutline : MonoBehaviour
+    {
+        /// <summary>
+        /// The material used to render the outline. Outline materials should normal have "Depth Write" set to Off and "Vertex Extrusion" enabled.
+        /// Most MRTK/Standard features should work as an outline material, but it is recommended to use a simple unlit material.
+        /// </summary>
+        public Material OutlineMaterial
+        {
+            get { return outlineMaterial; }
+            set
+            {
+                if (outlineMaterial != value)
+                {
+                    outlineMaterial = value;
+                    ApplyOutlineMaterial();
+                }
+            }
+        }
+
+        [Tooltip("The material used to render the outline. Outline materials should normal have \"Depth Write\" set to Off and \"Vertex Extrusion\" enabled.")]
+        [SerializeField]
+        protected Material outlineMaterial = null;
+
+        /// <summary>
+        /// How thick (in meters) should the outline be. Overrides the "Extrusion Value" in the MRTK/Standard material.
+        /// </summary>
+        public float OutlineWidth
+        {
+            get { return outlineWidth; }
+            set
+            {
+                if (outlineWidth != value)
+                {
+                    outlineWidth = value;
+                    ApplyOutlineWidth();
+                }
+            }
+        }
+
+        [Tooltip("How thick (in meters) should the outline be. Overrides the \"Extrusion Value\" in the MRTK/Standard material.")]
+        [SerializeField]
+        [Range(0.001f, 1.0f)]
+        protected float outlineWidth = 0.01f;
+
+        #region MonoBehaviour Implementation
+
+        /// <summary>
```

Comment:
I remember @Railboy recently removed a bunch of code that was using OnValidate incorrectly #4899. Wanted to double check that this was appropriate usage. It seems like we are actually using OnValidate here as a proxy to "when value changes, show changes in editor". Is there another way to do this, maybe with the inspector?

Reply:
I've heard some of the (warranted) complaints about using OnValidate. But, I do believe this is the intended use of it. The MRTK was using it for initialization in a few places, which is a dangerous pattern. I think it's perfectly acceptable to use OnValidate to respond to edit time property changes.